### DISCUSSION TITLE
is there any way to edit files in minIO directly?

### DISCUSSION THREAD

**Discussion by fengxueyoung** [minio/minio#17166] [Q&A] [answered]
im working on an online editing function, users can edit their files online on browsers, and save it after they done editing.
now our solution is: get the files from minIO, deliver it to the front end, and save the edited file to minIO
but considering the file might be big, even if user just change a word, we need to save the whole file again
is there any way that allow user to edit files and save them in minIO directly without going through my program?like api or plugins

**Comment by klauspost** [comment, marked as answer]
The S3 API does not allow partial mutations of objects, so you will need to download+modify+upload the objects.
